lib/manager/gradle/index.ts

```diff
@@ -25,50 +27,59 @@ import { platform } from '../../platform';
 import { LANGUAGE_JAVA } from '../../constants/languages';
 import * as datasourceMaven from '../../datasource/maven';
 import { DatasourceError } from '../../datasource';
+import { BinarySource } from '../../util/exec/common';
 
 export const GRADLE_DEPENDENCY_REPORT_OPTIONS =
   '--init-script renovate-plugin.gradle renovate';
 const TIMEOUT_CODE = 143;
 
-async function canExecute(path: string): Promise<boolean> {
-  try {
-    await access(path, constants.X_OK);
-    return true;
-  } catch {
-    return false;
+function gradleWrapperFileName(config: ExtractConfig): string {
+  if (
+    os.platform() === 'win32' &&
+    config.binarySource !== BinarySource.Docker
+  ) {
+    /* istanbul ignore next */
+    return 'gradlew.bat';
   }
+  /* istanbul ignore next */
```

Comment:
And this?
```suggestion
```

Reply:
This one is not actually necessary, true. I removed it.